ProjectStealth/Assets/Scripts/Character/GrapplingHook.cs

```diff
@@ -0,0 +1,251 @@
+        if ( hit_good_surface )
+        {
+            BoxCollider2D box = hit.collider.GetComponent<BoxCollider2D>();
+            if ( box != null )
+            {
+                target = box.bounds.center;
+                distance_to_target = Vector2.Distance( character_position, box.bounds.center );
+            }
+        }
+
+        if      ( hit_good_surface ) { StartGrapple( target ); }
+        else if ( hit_enemy )        { GetOverHere(); }
+        else if ( hit_bad_surface )  { NoisilyBounceOffWall( target ); }
+    }
+
+    /// <summary>
+    /// Starts the process of pulling yourself to the grapple point.
+    /// </summary>
+    /// <param name="target"></param>
+    private void StartGrapple( Vector2 target )
+    {
+        char_stats.current_master_state = CharEnums.MasterState.RappelState; // turn off default move behaviour.
+        this.target = target;
+        is_on = true;
+        is_retracting = false;
+
+        Vector3 character_position_3D = char_stats.char_collider.bounds.center;
+        Vector2 character_position = new Vector2( character_position_3D.x, character_position_3D.y );
+        angle = Mathf.Atan2( target.y - character_position.y, target.x - character_position.x );
+        Quaternion quaternion = Quaternion.Euler( 0.0f, 0.0f, angle * Mathf.Rad2Deg );
+        distance_to_target = Vector2.Distance(target, character_position); 
+        distance_from_target = 0.0f;
+
+        hook_instance  = GameObject.Instantiate( hook_prefab,  character_position, quaternion );
+        chain_instance = GameObject.Instantiate( chain_prefab, character_position, quaternion );
+    }
+
+    /// <summary>
+    /// Fires the hookshot, and FAILS.
+    /// Can be used to draw attention, though.
+    /// </summary>
+    private void NoisilyBounceOffWall( Vector2 target )
+    {
+        // makes a dink noise and sparks
+        did_fail = true;
+        StartGrapple( target );
+    }
+
+    /// <summary>
+    /// Pulls the enemy to you... unless it's heavy. Or ungrappleable.
+    /// </summary>
+    private void GetOverHere()
+    {
+        // Is enemy hookshottable + pullable?
+        bool is_hookshottable = true;
+        if ( ! is_hookshottable ) { return; }
+        bool is_pullable = false;
+        if ( ! is_pullable ) { ComingOverThere(); return; }
+
+        // pull the enemy!
+        // complications: enemies can move, need to stun during pull, but also maybe before so you can hit them? Or defer the pull check?
+        // TODO:
+    }
+
+    /// <summary>
+    /// Pulls you to a heavy enemy.
+    /// </summary>
+    private void ComingOverThere()
+    {
+        // pull
+        // complications: enemies can move, probably need to stun during pull, also maybe before so you can hit them? Or defer the check?
+        // TODO:
+        // StartGrapple( target );
+    }
+
+    /// <summary>
+    /// Makes the grappling hook shot function.
+    /// </summary>
+    private void Function()
```

Comment:
pretty generic name, but I guess it's not too hard to get confused in a small module